facia/app/controllers/FaciaController.scala

```diff
@@ -200,6 +202,11 @@ trait FaciaController
       case None => Future.successful(None)
     }
 
+    val networkFrontEdition = Edition.allWithBetaEditions.find(_.networkFrontId == path)
+    val deeplyRead = if (Switches.DeeplyReadSwitch.isSwitchedOn) {
+      networkFrontEdition.map(deeplyReadAgent.getTrails)
+    } else None
```

Comment:
Thanks @OllysCoding for looking. Here at this point we don't yet have access to the faciaPage object, because at this point we have futureFaciaPage which has type `Future[Option[(PressedPage, Boolean)]]`. After this point, we are pattern matching the futureFaciaPage and that's where we can use `faciaPage.isNetworkFront`

But because we need the deeplyRead data on 2 of the pattern match cases, I decided to get the list before we start the pattern match, to avoid code duplication. 

But maybe you mean that we can also check if it's a network front at that point? Something like this:
```
        withVaryHeader(
          remoteRenderer.getFront(
            ws = ws,
            page = faciaPage,
            pageType = pageType,
            mostViewed = mostViewedAgent.mostViewed(Edition(request)),
            mostCommented = mostViewedAgent.mostCommented,
            mostShared = mostViewedAgent.mostShared,
            deeplyRead = if (faciaPage.isNetworkFront) deeplyRead else None,
          )(request),
          targetedTerritories,
        )
```

Reply:
Makes sense, thank you for responding! I think it'll be okay as-is :)